### PR TITLE
Empty state documentation

### PR DESCRIPTION
**Pull request by ashharrison90**
- initial start at some empty state documentation
- what to do about the `Empty state` pattern page? do we need this anymore? 🤔

### CODE REVIEW

File: docs/05-components/empty-state.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+### `variant="call-to-action"`
+
+Use when there is no data to display and you want to encourage the user to take an action. Usually either to complete some initial configuration, or to create an item.
+
+```tsx
+import { EmptyState, LinkButton, TextLink } from '@grafana/ui';
+
+<EmptyState
+  variant="call-to-action"
```
Comment (joshhunt) [inline docs/05-components/empty-state.mdx:31]: Add the `live` tag after the tsx language specifier (` \`\`\`tsx live `) and make this an interactive demo ✨ 

You can remove the import statement as well.

Comment (ashharrison90) [reply docs/05-components/empty-state.mdx:31]: don't think i can do that until it's released in `@grafana/ui` though 😬

Comment (joshhunt) [reply docs/05-components/empty-state.mdx:31]: Ahh true. You can bump grafana-ui to canary version then?

**Review by joshhunt — COMMENTED** [review]
Good docs 👍

File: docs/05-components/empty-state.mdx (hunk 1 above)
Comment (ashharrison90) [reply docs/05-components/empty-state.mdx:31]: if we're happy with that then yeah i can do that 👍

Comment (ashharrison90) [reply docs/05-components/empty-state.mdx]: have updated to canary and added the live version of `not-found` (it's the only one that exists currently - waiting for an svg for the other variant 😭 )

**Comment by staton-hyse11** [comment]
This content looks great! 👏  

To the above question about what to do with the pattern page, I actually think this content is more appropriate for the patterns section. I know this is a "component," but what we are really standardizing here is the pattern for empty states across Grafana. This component is a more complex organism-level component outside of the core atom/molecule documents.

What are your thoughts on moving this to the patterns section?

**Review by staton-hyse11 — APPROVED** [review]
Regardless of the location for this, I think this is great to move forward with 👍

**Comment by ashharrison90** [comment]
> This content looks great! 👏
> 
> To the above question about what to do with the pattern page, I actually think this content is more appropriate for the patterns section. I know this is a "component," but what we are really standardizing here is the pattern for empty states across Grafana. This component is a more complex organism-level component outside of the core atom/molecule documents.
> 
> What are your thoughts on moving this to the patterns section?

sounds good to me, have moved it to Patterns 👍

**Pull request merged**